test/bats/test.bats

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env bats
+
+load helpers
+
+TESTS_DIR=library
+BATS_TESTS_DIR=test/bats
+WAIT_TIME=300
+SLEEP_TIME=5
+CLEAN_CMD="echo cleaning..."
+
+teardown() {
+  bash -c "${CLEAN_CMD}"
+  kubectl delete constrainttemplate --all
+}
+
+setup() {
+  kubectl config set-context --current --namespace default
+}
+
+@test "gatekeeper-controller-manager is running" {
+  wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "kubectl -n gatekeeper-system wait --for=condition=Ready --timeout=60s pod -l control-plane=controller-manager"
+}
+
+@test "gatekeeper-audit is running" {
+  wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "kubectl -n gatekeeper-system wait --for=condition=Ready --timeout=60s pod -l control-plane=audit-controller"
+}
+
+@test "namespace label webhook is serving" {
+  cert=$(mktemp)
+  CLEAN_CMD="${CLEAN_CMD}; rm ${cert}"
+  wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "get_ca_cert ${cert}"
+
+  kubectl run temp --generator=run-pod/v1  --image=tutum/curl -- tail -f /dev/null
+  kubectl wait --for=condition=Ready --timeout=60s pod temp
+  kubectl cp ${cert} temp:/cacert
+
+  wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "kubectl exec -it temp -- curl -f --cacert /cacert --connect-timeout 1 --max-time 2  https://gatekeeper-webhook-service.gatekeeper-system.svc:443/v1/admitlabel"
+  kubectl delete pod temp
+}
+
+@test "constrainttemplates crd is established" {
```

Comment:
Should we be using something like gatekeeper-validate here so we don't need to:

* spin up a k8s cluster
* install Gatekeeper
* check for eventual consistency
* deal with cleanup

?

Reply:
Thought around these two approaches is that kind is more representative of real-world usage.

Reply:
Fair enough. If the complexity gets to be too much, we can look at switching later.

Sorry for the late reply :/